.github/workflows/ci-emscripten.yml

```diff
@@ -0,0 +1,155 @@
+          curl -L https://ftp.gnu.org/gnu/gmp/gmp-6.3.0.tar.xz -o gmp-6.3.0.tar.xz
+          tar -xf gmp-6.3.0.tar.xz
+
+          cd gmp-6.3.0
+
+          emconfigure ./configure \
+            --disable-dependency-tracking \
+            --host none \
+            --disable-shared \
+            --enable-static \
+            --enable-cxx \
+            --prefix=${{ env.WASM_LIBRARY_DIR }}
+          emmake make -j $(nproc)
+          emmake make install
+
+      - name: Build libmpfr
+        if: steps.cache-wasm-library-dir.outputs.cache-hit != 'true'
+        env:
+          CFLAGS: "-fPIC"
+          WASM_LIBRARY_DIR: ${{ github.workspace }}/wasm-library-dir
+        run: |
+          curl -L https://ftp.gnu.org/gnu/mpfr/mpfr-4.2.1.tar.xz -o mpfr-4.2.1.tar.xz
+          tar -xf mpfr-4.2.1.tar.xz
+
+          cd mpfr-4.2.1
+
+          emconfigure ./configure \
+            --disable-dependency-tracking \
+            --disable-shared \
+            --with-gmp="${{ env.WASM_LIBRARY_DIR }}" \
+            --prefix=${{ env.WASM_LIBRARY_DIR }}
+          emmake make -j $(nproc)
+          emmake make install
+
+      # might need patch
+      - name: Build flint
+        if: steps.cache-wasm-library-dir.outputs.cache-hit != 'true'
+        env:
+          CFLAGS: "-fPIC"
+          WASM_LIBRARY_DIR: ${{ github.workspace }}/wasm-library-dir
+        run: |
+          curl -L https://github.com/flintlib/flint/releases/download/v3.2.0-rc1/flint-3.2.0-rc1.tar.xz -o flint-3.2.0-rc1.tar.xz
+          tar -xf flint-3.2.0-rc1.tar.xz
+
+          cd flint-3.2.0-rc1
+
+          emconfigure ./configure \
+            --disable-dependency-tracking \
+            --disable-shared \
+            --prefix=${{ env.WASM_LIBRARY_DIR }} \
+            --with-gmp=${{ env.WASM_LIBRARY_DIR }} \
+            --with-mpfr=${{ env.WASM_LIBRARY_DIR }}
+            --host=wasm32-unknown-emscripten
+            --disable-assembly
+          emmake make -j $(nproc)
+          emmake make install
+
+      - name: Persist WASM library directory to cache
+        uses: actions/cache/save@d4323d4df104b026a6aa633fdb11d772146be0bf # v4.2.2
+        with:
+          path: ${{ github.workspace }}/wasm-library-dir
+          key: wasm-library-dir-${{ hashFiles('.github/workflows/ci-emscripten.yml') }}
+
+      - name: Build python-flint
+        env:
+          WASM_LIBRARY_DIR: ${{ github.workspace }}/wasm-library-dir
+        run: |
+          export PKG_CONFIG_PATH="${{ env.WASM_LIBRARY_DIR }}/lib/pkgconfig:${PKG_CONFIG_PATH}"
+          export CFLAGS="-I${{ env.WASM_LIBRARY_DIR }}/include ${CFLAGS:-}"
+          export LDFLAGS="-L${{ env.WASM_LIBRARY_DIR }}/lib -lflint -lmpfr -lgmp ${LDFLAGS:-}"
+
+          echo "PKG_CONFIG_PATH=${PKG_CONFIG_PATH}"
+          echo "CFLAGS=${CFLAGS}"
+          echo "LDFLAGS=${LDFLAGS}"
+
+          pkg-config --modversion python3
+          pkg-config --modversion mpfr
+          pkg-config --modversion flint
+
+          pyodide build
```

Comment:
I'm not sure what arguments need to be passed here but we need to tell meson to allow other flint versions with `-Dflint_version_check=false`.

Reply:
`pyodide build` can pass config settings to pass on to the build backend. I added it in 382cdecda85f81b07a71488fbacdea08621b49ad.